### PR TITLE
Feat: add base structure for K8s pipeline

### PR DESCRIPTION
**Pull request by Tailp**
Hi @monperrus @gluckzhang :), final part of the base structure for K8s pipeline. Awaiting for comments.

### CODE REVIEW

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java [hunk 1]
```diff
@@ -428,7 +428,14 @@ public void mainProcess() {
 
     public static void main(String[] args) throws JSAPException {
         Launcher launcher = new Launcher(args);
-        launcher.mainProcess();
+        boolean kubernetesMode = false;
+        if (kubernetesMode) {
+            LOGGER.warn("We are now running with kubernetes mode");
+            PipelineBuildListener pipelineBuildListener = new PipelineBuildListener(launcher);
```
Comment (gluckzhang) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java]: Use the interface to declare the variable.

**Comment by Tailp** [comment]
Resolved :) . Awaiting for new comments.

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java [hunk 2]
```diff
@@ -428,7 +429,14 @@ public void mainProcess() {
 
     public static void main(String[] args) throws JSAPException {
         Launcher launcher = new Launcher(args);
-        launcher.mainProcess();
+        boolean kubernetesMode = false;
+        if (kubernetesMode) {
```
Comment (monperrus) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/Launcher.java]: avoid if/then/else. use polymorphism instead.

**Comment by Tailp** [comment]
@monperrus  resolved :) .

**Pull request merged**